api/ussd/engine.js

```diff
@@ -397,8 +399,14 @@ module.exports = ({
                 return await new Promise((resolve, reject) => {
                     parser.onend = () => {
                         resolve({
-                            shortMessage: shortMessage.join(''),
-                            sourceAddr: data.system.phone
+                            ...({
+                                shortMessage: shortMessage.join(''),
+                                sourceAddr: data.system.phone
+                            }),
+                            ...(
+                                !exposeState && {} ||
+                                {state: data}
+                            )
```

Comment:
this is overcomplicated, consider changing it to:
```js
{
    shortMessage: shortMessage.join(''),
    sourceAddr: data.system.phone,
    ...exposeState && {state: data}
}
```

Reply:
didnt try spread of undefined :)